lib/oxide-vpc/src/engine/geneve.rs

```diff
@@ -5,7 +5,72 @@
 // Copyright 2025 Oxide Computer Company
 
 //! Geneve option types specific to the Oxide VPC dataplane.
-
+//!
+//! # Oxide Geneve Options
+//!
+//! This module defines Geneve options used in the Oxide rack network to carry
+//! VPC-specific metadata during packet encapsulation. All options use the Oxide
+//! option class (`GENEVE_OPT_CLASS_OXIDE` = 0x0129).
+//!
+//! ## Option Types
+//!
+//! - **External** (0x00): Indicates a packet originated from outside the rack
+//!   and was encapsulated by the switch NAT ingress path with Geneve wrapping.
+//!   OPTE decapsulates before delivering to the guest.
+//! - **Multicast** (0x01): Carries multicast replication strategy as a 2-bit
+//!   field for coordinating delivery between OPTE and sidecar switch logic.
+//! - **Mss** (0x02): Carries original TCP MSS for MSS clamping/boosting to
+//!   prevent MTU issues during underlay encapsulation.
+//!
+//! ## Multicast Option Encoding
+//!
+//! The multicast option uses a compact 2-bit encoding aligned with sidecar.p4's
+//! processing constraints:
+//!
+//! ```text
+//! Option body (4 bytes):
+//! ┌──────────┬────────────────────────────┐
+//! │ Bits 7-6 │ Bits 5-0 + remaining bytes │
+//! │ (u2)     │ (reserved, must be 0)      │
+//! └──────────┴────────────────────────────┘
+//!    │
+//!    └─> Replication mode:
+//!        00 = External (local guest delivery)
+//!        01 = Underlay (infrastructure forwarding)
+//!        10 = All (both External and Underlay)
+//!        11 = Reserved
+//! ```
+//!
+//! ### Replication Semantics
+//!
+//! - **External**: Packet should be decapsulated and delivered to local guest
+//!   instances subscribed to this multicast group. Switch sets `nat_egress_hit`
+//!   to trigger decapsulation before delivery.
+//! - **Underlay**: Packet should remain encapsulated and forwarded to underlay
+//!   infrastructure destinations.
+//! - **All**: Bifurcated delivery to both local guests (decapsulated) and
+//!   underlay destinations (encapsulated).
```

Comment:
Hmm, still of the opinion that OPTE should only care about these on Tx, for telling the switch how to replicate the frame. Same comment about `External` meaning what we would probably consider 'internal' delivery.